blocks/table/src/app.tsx

```diff
@@ -0,0 +1,305 @@
+  "https://blockprotocol-gkgdavns7.stage.hash.ai/@luisbett/types/property-type/title/";
+const localColumnsKey: RootKey =
+  "https://blockprotocol-hk4sbmd9k.stage.hash.ai/@yusuf123/types/property-type/table-local-column/";
+const localRowsKey: RootKey =
+  "https://blockprotocol-hk4sbmd9k.stage.hash.ai/@yusuf123/types/property-type/table-local-row/";
+const isStripedKey: RootKey =
+  "https://blockprotocol-hk4sbmd9k.stage.hash.ai/@yusuf123/types/property-type/is-striped/";
+const hideHeaderRowKey: RootKey =
+  "https://blockprotocol-hk4sbmd9k.stage.hash.ai/@yusuf123/types/property-type/hide-header-row/";
+const hideRowNumbersKey: RootKey =
+  "https://blockprotocol-hk4sbmd9k.stage.hash.ai/@yusuf123/types/property-type/hide-row-numbers/";
+
+const columnTitleKey: ColumnKey =
+  "https://blockprotocol-gkgdavns7.stage.hash.ai/@luisbett/types/property-type/title/";
+const columnIdKey: ColumnKey =
+  "https://blockprotocol-gqpc30oin.stage.hash.ai/@nate/types/property-type/id/";
+
+const emptySelection = {
+  columns: CompactSelection.empty(),
+  rows: CompactSelection.empty(),
+};
+
+export const App: BlockComponent<RootEntity> = ({
+  graph: { blockEntitySubgraph, readonly },
+}) => {
+  if (!blockEntitySubgraph) {
+    throw new Error("No blockEntitySubgraph provided");
+  }
+
+  const blockRootRef = useRef<HTMLDivElement>(null);
+  const { graphModule } = useGraphBlockModule(blockRootRef);
+
+  const { rootEntity: blockEntity } = useEntitySubgraph<
+    RootEntity,
+    RootEntityLinkedEntities
+  >(blockEntitySubgraph);
+
+  const {
+    metadata: {
+      recordId: { entityId: blockEntityId },
+      entityTypeId: blockEntityTypeId,
+    },
+    properties: {
+      [titleKey]: title = "",
+      [localColumnsKey]: localColumns = [],
+      [localRowsKey]: localRows = [],
+      [isStripedKey]: isStriped = false,
+      [hideHeaderRowKey]: hideHeaderRow = false,
+      [hideRowNumbersKey]: hideRowNumbers = false,
+    },
+  } = blockEntity;
+
+  const rows = localRows;
+  const [selection, setSelection] = useState<GridSelection>(emptySelection);
+
+  const [headerMenu, setHeaderMenu] = useState<{
+    col: number;
+    bounds: Rectangle;
+  }>();
+
+  const addNewColumn = () => {
+    return updateEntity({
+      [localColumnsKey]: [
+        ...localColumns,
+        {
+          [columnIdKey]: String(Date.now()),
+          [columnTitleKey]: `Column ${localColumns.length + 1}`,
+        },
+      ],
+    });
+  };
+
+  const addNewRow = async () => {
+    const newRows = [...rows, {}];
+
+    return updateEntity({ [localRowsKey]: newRows });
+  };
+
+  // this should definitely be a sync function
+  const handleCellsEdited: DataEditorProps["onCellsEdited"] = (newValues) => {
+    const newRows = produce(rows, (draftRows) => {
+      for (const { value, location } of newValues) {
+        const [colIndex, rowIndex] = location;
+
+        const columnId = columns[colIndex]?.id;
+
+        if (columnId) {
+          // @ts-ignore
```

Comment:
if we `expect-error` we will be warned if this line becomes unnecessary – also suggest mentioning what error we are expecting in case more appear when someone revisits this
```suggestion
          // @ts-expect-error -- type instantiation is deep and possibly infinite
```